### DISCUSSION TITLE
Accessing Document vectors and computing similarity is too slow without batching

### DISCUSSION THREAD

**Discussion by ATAboukhadra** [explosion/spaCy#8676] [Help: Coding & Implementations] [answered]
Hi,

I use this library and other Spacy models to create Doc objects.
I use the pipe() method to apply this to a large corpus of text. The main challenge is that accessing the vector of each document is too slow.
Is there a way to get only the vector from applying the model on the text? Or extract the vectors in batches as well?
This problem was raised from the problem of similarity where I couldn't also use the method similarity() on batches but only 1 by 1. Is there a way to compute similarity in batches?

I'm using a 4-core CPU.

I hope my question is clear.

Thanks.

**Comment by polm** [comment, marked as answer]
> Is there a way to get only the vector from applying the model on the text? Or extract the vectors in batches as well?

As mentioned in the speed FAQ in #8402, you can turn off components you aren't using. For the document vectors you just need the tok2vec (or Transformer), so you can turn off everything else in the pipeline. If you're using `nlp.pipe` that's already going to be doing as much batching as can be done, though you can tweak the parameters to it.

> This problem was raised from the problem of similarity where I couldn't also use the method similarity() on batches but only 1 by 1. Is there a way to compute similarity in batches?

For finding similarity with many items you may want to look at libraries that specifically support vector search. [Annoy](https://github.com/spotify/annoy) is easy to use and probably a good starting point.
